packages/server/graphql/public/types/User.ts

```diff
@@ -166,6 +168,18 @@ const User: UserResolvers = {
       .sort((a, b) => (a.sortOrder > b.sortOrder ? -1 : 1))
 
     return connectionFromTemplateArray(allActivities, first, after)
+  },
+  parseSAMLMetadata: async (_source, {metadata, domain}) => {
```

Comment:
-1 see before, this really feels more like a mutation although we're not modifying the server state here. But we're using this to start the verification flow which in the end stores the new data

Reply:
i'm interested in hearing more about how we should define if something is a query or mutation! This is a really good use case because the client is using a query like a mutation. IMO, just because the client does something a bit odd doesn't mean the server should change how it's defined, but I want to know more!

from a GraphQL perspective, the only real difference between a query & mutation is if a request payload includes multiple mutations, they must run sequentially, whereas multiple queries will run in parallel. In advanced GraphQL setups, mutations are run against the primary DB, whereas queries are run against read replicas in order to decrease load on the primary. TBH, it's not a big deal in our app, so whatever we decide, I just want to make sure we stay consistent.